### PR TITLE
Make sure that EraseAll moves the Terminal viewport

### LINKED ISSUE #2832: Feature Request Move scroll buffer instead of clearing on CTRL+L and on clear
# Description of the new feature/enhancement
On WSL, Typing `clear` or pressing `CTRL+L` clears the scroll buffer history while most Linux terminals(both real ones like gnome-terminal, and virtual ones like tmux) move the scroll history instead of clearing it. Can we use same semantics for Windows Terminal?

### PR DESCRIPTION
**Pull request by zadjii-msft**
The Erase All VT sequence (`^[[2J`) is supposed to erase the entire
contents of the viewport. The way it usually does this is by shifting
the entirety of the viewport contents into scrollback, and starting the
new viewport below it. 

Currently, conpty doesn't propagate that state change correctly. When
conpty gets a 2J, it simply erases the content of the connected
terminal's viewport, by writing over it with spaces. Conpty didn't
really have a good way of communicating "your viewport should move", it
only knew "the buffer is now full of spaces".

This would lead to bugs like #2832, where pressing <kbd>ctrl+L</kbd> in
`bash` would delete the current contents of the viewport, instead of
moving the viewport down.

This PR makes sure that when conpty sees a 2J, it passes that through
directly to the connected terminal application as well. Fortunately, 2J
was already implemented in the Windows Terminal, so this actually fixes
the behavior of <kbd>ctrl+L</kbd>/`clear` in WSL in the Terminal.

## References

* #4252 - right now this isn't the _most_ optimal scenario, we're
  literally just printing a 2J, then we'll perform "erase line" `height`
  times. The erase line operations are all redundant at this point - the
  entire viewport is blank, but conpty doesn't really know that.
  Fortunately, #4252 was already filed for me to come through and
  optimize this path.

## PR Checklist
* [x] Closes #2832
* [x] I work here
* [x] Tests added/passed
* [n/a] Requires documentation to be updated

## Validation Steps Performed
* ran tests
* compared <kbd>ctrl+L</kbd> with its behavior in conhost
* compared `clear` with its behavior in conhost

### CODE REVIEW

**Comment by DHowett-MSFT** [comment]
I’m surprised we didn’t just return false at adaptDispatch. This does seem more _robust_ though.

**Comment by zadjii-msft** [comment]
> I’m surprised we didn’t just return false at adaptDispatch

derp that would have been easier

File: src/host/getset.cpp [hunk 1]
```diff
@@ -1641,9 +1641,18 @@ void DoSrvPrivateEnableAlternateScroll(const bool fEnable)
 //  The ScreenBuffer to perform the erase on.
 // Return value:
 // - STATUS_SUCCESS if we succeeded, otherwise the NTSTATUS version of the failure.
```
Comment (carlos-zamora) [inline src/host/getset.cpp:1643]: nit: update comment on return value

File: src/renderer/vt/state.cpp [hunk 2]
```diff
@@ -509,3 +509,17 @@ void VtEngine::SetResizeQuirk(const bool resizeQuirk)
...
+//   client requests a Erase All operation, we need to manually tell the
+//   connected terminal to do the same thing, so that the terminal will move
+//   it's own buffer contents into the scrollback.
+// Arguments:
+// - <none>
+// Return Value:
+// - S_OK if we wrote the sequences successfully, otherwise an appropriate HRESULT
+[[nodiscard]] HRESULT VtEngine::ManuallyClearViewport() noexcept
+{
+    return _ClearScreen();
```
Comment (carlos-zamora) [inline src/renderer/vt/state.cpp:524]: Should we just skip the middle-man and make `ClearScreen` public?

File: src/cascadia/UnitTests_TerminalCore/ConptyRoundtripTests.cpp [hunk 3]
```diff
@@ -2654,6 +2767,18 @@ void ConptyRoundtripTests::ClsAndClearHostClearsScrollbackTest()
                                                                      { 0, 0 },
                                                                      cellsWritten));
     }
+    else if (clearBufferMethod == ClearWithVT)
+    {
+        // DebugBreak();
```
Comment (DHowett-MSFT) [inline src/cascadia/UnitTests_TerminalCore/ConptyRoundtripTests.cpp:2772]: :O

File: src/renderer/vt/state.cpp (hunk 2 above)
Comment (zadjii-msft) [reply src/renderer/vt/state.cpp]: Eh, I just got rid of this entirely

**Pull request merged**